src/drivers/distance_sensor/pga460/pga460.cpp

```diff
@@ -152,34 +150,39 @@ int PGA460::initialize_thresholds()
 	// Must wait >50us per datasheet.
 	usleep(100);
 
-	if (read_threshold_registers()) {
-		return 1;
+	if (read_threshold_registers() == PX4_OK) {
+		return PX4_OK;
 
 	} else {
 		print_device_status();
-		return 0;
+		return PX4_ERROR;
 	}
 }
 
 uint32_t PGA460::collect_results()
 {
-	px4_pollfd_struct_t fds[1];
-	fds[0].fd = _fd;
-	fds[0].events = POLLIN;
 
-	int timeout = 10;
-	uint8_t buf_rx[6] = {0};
+	uint8_t buf_rx[6] = {};
+	int bytes_available = sizeof(buf_rx);
+	int total_bytes = 0;
 
-	int ret = px4_poll(&fds[0], (sizeof(fds) / sizeof(fds[0])), timeout);
+	do {
+		int ret = px4_read(_fd, buf_rx + total_bytes, sizeof(buf_rx));
```

Comment:
Use the real ::read, ::write, ::close, etc on the serial port file descriptor. The `px4_` versions are for working with virtual files on linux and qurt. On Nuttx `px4_read == read`. Yes it's confusing...